apps/server/src/modules/deletion/uc/deletion-request.uc.ts

```diff
@@ -0,0 +1,216 @@
+	targetRef: { targetRefDoamin: DeletionDomainModel; targetRefId: EntityId };
+	deletionPlannedAt: Date;
+	statistics?: DeletionLogStatistic[];
+}
+
+export interface DeletionLogStatistic {
+	domain: DeletionDomainModel;
+	modifiedCount?: number;
+	deletedCount?: number;
+}
+
+export interface DeletionRequestProps {
+	targetRef: { targetRefDoamin: DeletionDomainModel; targetRefId: EntityId };
+	deleteInMinutes?: number;
+}
+
+@Injectable()
+export class DeletionRequestUc {
+	constructor(
+		private readonly deletionRequestService: DeletionRequestService,
+		private readonly deletionLogService: DeletionLogService,
+		private readonly accountService: AccountService,
+		private readonly classService: ClassService,
+		private readonly courseGroupService: CourseGroupService,
+		private readonly courseService: CourseService,
+		private readonly filesService: FilesService,
+		private readonly lessonService: LessonService,
+		private readonly pseudonymService: PseudonymService,
+		private readonly teamService: TeamService,
+		private readonly userService: UserService
+	) {}
+
+	async createDeletionRequest(
+		deletionRequest: DeletionRequestProps
+	): Promise<{ requestId: EntityId; deletionPlannedAt: Date }> {
+		const result = await this.deletionRequestService.createDeletionRequest(
+			deletionRequest.targetRef.targetRefId,
+			deletionRequest.targetRef.targetRefDoamin,
+			deletionRequest.deleteInMinutes
+		);
+
+		return result;
+	}
+
+	async executeDeletionRequests(limit?: number): Promise<void> {
+		const deletionRequestToExecution: DeletionRequest[] = await this.deletionRequestService.findAllItemsToExecute(
+			limit
+		);
+
+		for (const req of deletionRequestToExecution) {
+			// eslint-disable-next-line no-await-in-loop
+			await this.executeDeletionRequest(req);
+		}
+	}
+
+	async findById(deletionRequestId: EntityId): Promise<DeletionRequestLog> {
+		const deletionRequest: DeletionRequest = await this.deletionRequestService.findById(deletionRequestId);
+		let response: DeletionRequestLog = {
+			targetRef: {
+				targetRefDoamin: deletionRequest.targetRefDomain,
+				targetRefId: deletionRequest.targetRefId,
+			},
+			deletionPlannedAt: deletionRequest.deleteAfter,
+		};
+
+		if (deletionRequest.status === DeletionStatusModel.SUCCESS) {
+			const deletionLog: DeletionLog[] = await this.deletionLogService.findByDeletionRequestId(deletionRequestId);
+			const deletionLogStatistic: DeletionLogStatistic[] = deletionLog.map((log) => {
+				return {
+					domain: log.domain,
+					modifiedCount: log.modifiedCount,
+					deletedCount: log.deletedCount,
+				};
+			});
+			response = { ...response, statistics: deletionLogStatistic };
+		}
+
+		return response;
+	}
+
+	async deleteDeletionRequestById(deletionRequestId: EntityId): Promise<void> {
+		await this.deletionRequestService.deleteById(deletionRequestId);
+	}
+
+	private async executeDeletionRequest(deletionRequest: DeletionRequest): Promise<void> {
+		await Promise.all([
```

Comment:
Please rewrite it to try chtch without then 
```suggestion
		[
					,
					classesUpdated,
					courseGroupUpdated,
					courseUpdated,
					fileDeleted,
					filesPermisionUpdated,
					lessonUpdated,
					pseudonymDeleted,
					teamsUpdated,
					userDeleted,
				] = await Promise.all([
```

Reply:
done